crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -880,17 +880,26 @@ impl LogsControllerInner {
         }
     }
 
-    fn on_partition_table_update(&mut self, partition_table: &PartitionTable) {
+    fn on_partition_table_update(
+        &mut self,
+        cluster_configuration: Option<Arc<ClusterConfiguration>>,
+        partition_table: &PartitionTable,
+    ) {
+        let Some(cluster_configuration) = cluster_configuration else {
+            return;
+        };
+
         // update the provisioning logs
         for (partition_id, _) in partition_table.partitions() {
+            let cluster_configuration = Arc::clone(&cluster_configuration);
             self.logs_state
                 .entry((*partition_id).into())
                 .or_insert_with(|| {
                     let log_id = (*partition_id).into();
-                    debug!(%log_id, "Try provisioning log with provider '{}'", self.default_provider);
+                    debug!(%log_id, "Try provisioning log with provider '{}'", cluster_configuration.default_provider);
                     LogState::Provisioning {
                         log_id,
-                        provider_kind: self.default_provider,
+                        cluster_configuration,
```

Comment:
What's the idea behind storing the current cluster configuration with the `Provisioning` state? Can't the cluster configuration change while we are trying to provision a log?

Reply:
it definitely can. I already removed it as part of the current changes. But good observation as usual ❤️